src/app/components/TradeModal.ts

```diff
@@ -0,0 +1,220 @@
+import { AnimalNames } from '~src/Enums/AnimalNamesEnum';
+import { Player } from '~src/Player';
+import { Herd } from '../logic/Herd';
+import { Offer, Trade } from '../Trade';
+import { Render } from '../utils/Render';
+import { EmptyModal } from './EmptyModal';
+
+export class TradeModal extends EmptyModal {
+  tradeForm: HTMLElement;
+  playerView: HTMLElement;
+  bankView: HTMLElement;
+  warning: HTMLElement;
+  backButton: HTMLElement;
+  player: Player;
+
+  constructor(private trade: Trade, firstPlayer: Player) {
+    super();
+    this.player = firstPlayer;
+    this.playerView = Render.elementFactory('div', {
+      className: 'trade__player-wrapper',
+    });
+    this.bankView = Render.elementFactory('div', {
+      className: 'trade__player-wrapper',
+    });
+    this.tradeForm = Render.elementFactory(
+      'form',
+      {
+        action: '',
+        method: 'post',
+        className: 'trade',
+      },
+      this.playerView,
+      Render.elementFactory('input', {
+        type: 'submit',
+        value: 'Trade',
+        className: 'trade__submit',
+      }),
+      this.bankView,
+    );
+    this.warning = Render.elementFactory('p', {
+      className: 'warning',
+    });
+    this.backButton = Render.elementFactory('button', {}, 'back');
+    Render.childrenInjector(
+      this.modalContainer,
+      this.tradeForm,
+      this.warning,
+      this.backButton,
+    );
+  }
+
+  /**
+   * Creates TradeModal and returns it as HTMLElement.
+   */
+  createModal(): HTMLElement {
+    Render.childrenInjector(
+      this.bankView,
+      this.createHerdView(this.trade.thisBank, true),
+    );
+    Render.childrenInjector(
+      this.playerView,
+      this.createHerdView(this.player),
+    );
+    this.tradeForm.addEventListener('submit', this.handleSubmit);
+    this.modal.addEventListener('keydown', this.clearWarning);
+    this.modal.addEventListener('click', this.clearWarning);
```

Comment:
I see that clearWarning is declared below, but I don't understand what should it do?